llvm/lib/CodeGen/AsmPrinter/OxCamlGCPrinter.cpp

```diff
@@ -173,10 +187,16 @@ bool OxCamlGCMetadataPrinter::emitStackMaps(Module &M, StackMaps &SM, AsmPrinter
 
     // frame_data
     uint64_t FrameSize = CSI.CSFunctionInfo.StaticStackSize;
-    if (CSI.ID != StatepointDirectives::DefaultStatepointID)
-      FrameSize += CSI.ID; // Stack offset from OxCaml
     FrameSize += PtrSize; // Return address
 
+    if (CSI.ID != StatepointDirectives::DefaultStatepointID) {
+      // Alloc bit
+      if (IDHasAlloc(CSI.ID)) FrameSize |= allocMask;
```

Comment:
It'd be better to first update the FrameSize, including the adjustment for the stackOffsetOfID in L197, and assert the bottom bits are 0, before masking.